### ISSUE TITLE
basic_string::_S_construct null not valid

### ISSUE THREAD

**Issue by rec3141** [rrwick/Filtlong#5]
oops I broke it. is it ok to run fasta instead of fastq?

```
~/apps/bin/filtlong -1 R1.fastq.gz -2 R2.fastq.gz --min_length 1000 --keep_percent 90 --target_bases 100000000 --trim --split 100 --mean_q_weight 10 Lib1_total.fa

Hashing 16-mers from Illumina reads
  R1.fastq.gz (296,289,686 bp)
  R2.fastq.gz (302,370,503 bp)
  2,328,592 reads, 10,036,530 16-mers

Scoring long reads
  840,680 reads (6,551,960,811 bp)
  after trimming and splitting: 7,794,202 reads (4,577,612,765 bp)

Filtering long reads
  target: 100,000,000 bp
  keeping 100,022,917 bp

Outputting passed long reads
@efd120a2-8b49-4da7-b50e-b4f83a59cd08_45-8301 runid=202b5ecbf0ff9176d3391f55aa069dcc609f99db read=8359 ch=394 start_time=2017-04-29T09:48:42Z
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_S_construct null not valid
Aborted

```

**Comment by rrwick** [comment]
At the moment, no. If you don't have Illumina reads, then FASTA input couldn't work because Filtlong would have no means to assess quality. But you _do_ have Illumina reads (as you do in your example), then FASTA input makes sense, it's just not implemented.

Planned enhancements:
1. Make FASTA inputs work if there are Illumina reads.
2. Make Filtlong give a nice error message if FASTA input is given without Illumina reads.

In the meantime, a crude workaround would be to convert your FASTA reads to FASTQ. Filtlong doesn't use quality score if Illumina reads are available, so it doesn't matter what you use. I'm not aware of a FASTA to FASTQ conversion tool, but I'm sure a bit of awk wizardry could do the trick.

**Comment by rrwick** [comment]
Okay, I've implemented this one in Filtlong's new version (0.2.0). It works as I said above: FASTA inputs are okay, but only if you use an external reference. Give it a try and let me know if you run into any problems.
